src/libraries/System.Linq/src/System/Linq/Sum.cs

```diff
@@ -49,6 +72,103 @@ public static partial class Enumerable
             return sum;
         }
 
+        private static T SumSignedIntegersVectorized<T>(ReadOnlySpan<T> span)
+            where T : struct, IBinaryInteger<T>, ISignedNumber<T>
+        {
+            Debug.Assert(span.Length >= Vector<T>.Count * 4);
+            Debug.Assert(Vector<T>.Count > 2);
+            Debug.Assert(Vector.IsHardwareAccelerated);
+
+            ref T ptr = ref MemoryMarshal.GetReference(span);
+            int length = span.Length;
+
+            // Overflow testing for vectors is based on setting the sign bit of the overflowTracking
+            // vector for an element if the following are all true:
+            //   - The two elements being summed have the same sign bit. If one element is positive
+            //     and the other is negative then an overflow is not possible.
+            //   - The sign bit of the sum is not the same as the sign bit of the previous accumulator.
+            //     This indicates that the new sum wrapped around to the opposite sign.
+            //
+            // By bitwise or-ing the overflowTracking vector for each step we can save cycles by testing
+            // the sign bits less often. If any iteration has the sign bit set in any element it indicates
+            // there was an overflow.
+            //
+            // Note: The overflow checking in this algorithm is only correct for signed integers.
+            // If support is ever added for unsigned integers then the overflow check should be:
+            //   overflowTracking |= (accumulator & data) | Vector.AndNot(accumulator | data, sum);
+
+            Vector<T> accumulator = Vector<T>.Zero;
+
+            // Build a test vector with only the sign bit set in each element. JIT will fold this into a constant.
+            Vector<T> overflowTestVector = new(T.RotateRight(T.MultiplicativeIdentity, 1));
```

Comment:
Given that we know `T` works with `Vector`, we know it must be a primitive value which is 2's complement. So we can add `where T : IMinMaxValue<T>` and simply use `T.MinValue` instead to get a mask of the `sign`.

Reply:
Done